src/files/DCDFile.cpp

```diff
@@ -0,0 +1,457 @@
+#include <cassert>
+#include <cstdint>
+#include <cstdlib>
+#include <fstream>
+#include <functional>
+#include <iostream>
+#include <string>
+
+#include "chemfiles/File.hpp"
+#include "chemfiles/files/DCDFile.hpp"
+
+#include "chemfiles/error_fmt.hpp"
+
+using namespace std;
+using namespace chemfiles;
+
+DCDFile::DCDFile(std::string path, File::Mode mode)
+{
+   const char* openmode;
+   if (mode == File::READ) {
+      openmode = "r";
+      fileStream.exceptions(ifstream::failbit);
+      try
+      {
+         fileStream.open(path, ios::in | ios::binary);
+      }
+      catch (ifstream::failure& e)
+      {
+         cerr << "Exception opening/reading file '" << path << "' : " << endl;
+         cerr << "Please chech the path of the file and if it exists." << endl;
+      }
+      //Read dcd file header 
+      readHeader();
+   }
+   else if (mode == File::WRITE) {
+      openmode = "w";
+      fileStream.exceptions(ofstream::failbit);
+      try
+      {
+         fileStream.open(path, ios::app | ios::binary);
+      }
+      catch (ofstream::failure& e)
+      {
+         cerr << "Exception creating file '" << path << "' : " << endl;
+         cerr << "Couldn't create the file." << endl;
+      }
+
+   }
+}
+
+void DCDFile::readHeader()
+{
+   unsigned int fortcheck1, fortcheck2;
+   long initialPos = fileStream.tellg();
+   //This is the trick for reading binary data from fortran file, see the method checkFortranIOerror for more details.
+   fileStream.read((char*)&fortcheck1, sizeof(unsigned int));               //consistency check 1
+   fileStream.read((char*)HDR, sizeof(char) * 4);                       //first data block written by fortran  : a character array of length 4.
+   fileStream.read((char*)ICNTRL, sizeof(int) * 20);                    //second data block written by fortran : an integer(4) array of length 20.
+   fileStream.read((char*)&fortcheck2, sizeof(unsigned int));         //consistency check 2
+
+   // if the 2 unsigned ints have a different value there was an error
+   if (!checkFortranIOerror(__FILE__, __LINE__, fortcheck1, fortcheck2))
+   {
+      return;
```

Comment:
This should be an exception instead of silently failing

Reply:
Absolutely, again I forgot to change it after debugging.